templates/slide.html

```diff
@@ -420,23 +435,44 @@ <h2 class="popover__title">{{disp_str[i]}}</h2>
  function onPrevPage() {
    // empty previous child elements of textLayer
    $("#textLayerid").empty();
-
+   course = "{{course_name}}";
+    lec = "{{lec_name}}";
    if (pageNum <= 1) {
      return;
    }
+  //Added updated_slide to keep track of slide name
    pageNum--;
+   new_slide_name= "{{course_name}}".concat("---").concat("{{lec_name}}").concat("---slide").concat((pageNum-1).toString());
+   logdata = JSON.stringify({
+        action: 'prev',
+        updated_slide: new_slide_name, 
+        route: window.location.pathname
+    });
+    navigator.sendBeacon('{{base_url}}log_action', logdata);
    queueRenderPage(pageNum);
    changedSlide(pageNum, pageNum+1);
  }
  document.getElementById('prev').addEventListener('click', onPrevPage);
 
  function onNextPage() {
    // if current pageNum requested is not valid, then return
-   // the following two lines is not required as the next will be disabled
+   // the following two lines is not required as the next will be disabled\
+    course = "{{course_name}}";
+    lec = "{{lec_name}}";
+    new_slide_name = course.concat("---").concat(lec).concat("---slide").concat((pageNum).toString());
```

Comment:
"{{course_name}}".concat("---").concat("{{lec_name}}").concat("---slide").concat((pageNum-1).toString());

Reply:
I have created the new log_action function and applied it onNextPage, PrevPage and hide and unhide logging. I saw there are a lot of other places where logging is used such as cache open. Should I call that function there too or are the four functions ok?

I wanted to find more about the url_path input, so I directly call {{base url}} in the log_action function for now. And I send in the {{base_url}} as the input for url_path parameter. I wasn't sure if this was the right thing to do, so wanted to make sure if this is ok.

There are also couple of more places where course= {{course_name}} and then new_slide_name = course.concat..... Should I change them or are they ok?

Thank you again for reviewing my changes and the detailed instructions. Please just let me know if I have to work on anything else.

Reply:
_I have created the new log_action function and applied it onNextPage, PrevPage and hide and unhide logging. I saw there are a lot of other places where logging is used such as cache open. Should I call that function there too or are the four functions ok?

I wanted to find more about the url_path input, so I directly call {{base url}} in the log_action function for now. And I send in the {{base_url}} as the input for url_path parameter. I wasn't sure if this was the right thing to do, so wanted to make sure if this is ok.

There are also couple of more places where course= {{course_name}} and then new_slide_name = course.concat..... Should I change them or are they ok?_

Reply:
I have created the new log_action function and applied it onNextPage, PrevPage and hide and unhide logging. I saw there are a lot of other places where logging is used such as cache open. Should I call that function there too or are the four functions ok? **Yes, use the log_action function everywhere. I suggest creating a new file logging.js and adding the function to the file. Then call this function log_action wherever logging is used. (Note you can use var base_url = $('#base-url').data('val'); to extract the base_url value, see explain.js Line 1)** 

I wanted to find more about the url_path input, so I directly call {{base url}} in the log_action function for now. And I send in the {{base_url}} as the input for url_path parameter. I wasn't sure if this was the right thing to do, so wanted to make sure if this is ok. **Remove the url_path parameter if not used. What you have right now should be fine.**

There are also couple of more places where course= {{course_name}} and then new_slide_name = course.concat..... Should I change them or are they ok? **Change them as well, we want to be consistent everywhere.**

Reply:
I changed the hide, unhide and close open logging to the newer version and also removed the url_path from log_action. I put the log_action function in a log.js file in the static folder and import that page in base.html. I also added a explanation_google_search.js file, it is mainly being used by explain.js for the googleQueryExplanation() function

Reply:
I have used the config.py to get the link for explanation_google_search.js, and I have shared the updated config file on slack. I get all three constants from the config.py file now. Thank you again for the example and the detailed instructions. Please just let me know if anything else comes up or if there is something that I need to fix.